Feature create score bk

idelcano opened the pull request:
Closes #2065 
@xurxodev  i only refactorized the ScoreDB and related calls but i don´t know if i need add more changes in this issue
(I started from : https://github.com/EyeSeeTea/malariapp/issues/1904 )

idelcano committed:
refactor scoreDB

idelcano committed:
fix some nulls

idelcano committed:
Merge branch 'development' into feature-create_score_bk

idelcano committed:
delete empty classes after merge

xurxodev reviewed (changes requested):
@idelcano There are some things to review:
- You have started this branch from #1904 but this is not necessary because you do not modify the pull process.
Please modify this issue to start from the development branch to avoid more bottlenecks with issue #1904 
- this issue has two parts: uid in the score table and uid in the survey table, the second part is pending, please implement it thanks

idelcano committed:
Merge branch 'development' into feature-create_score_bk

idelcano committed:
create surveyDB with eventuid on constructor

idelcano committed:
modify empty survey creation to generate the event uid

xurxodev committed:
Survey entity can not be coupled to dhis sdk:
- when we will need create new surveys from domain, we will create a
CreateNewSurveyUseCase and the request the new uid through of a
adapter with its abstraction to SDK (like repositories)

xurxodev reviewed (approved):
thanks @idelcano I have realized changes to we don´t couple to dhis sdk
